include/util.h

```diff
@@ -6,6 +6,15 @@
 #define WRITE_END 1
 #define EMAIL_BODY_LENGTH 4096
 
+#define CNC_MALLOC(SIZE)                  \
+	({                                \
+		void *ptr = malloc(SIZE); \
+		if (ptr == NULL) {        \
+			return -2;        \
```

Comment:
If we plan to use `-2` for all memory allocation errors, it's better to define a macro for it.

Reply:
Yes i believe from now on we should reserve `-2` for memory allocation, since it also lines up with
`ini_parse` from `inih` library which has the same value built-in for `malloc` error.

Reply:
Why not `-ENOMEM`, provided to us by the Linux kernel?

Reply:
We cannot change the value that `ini_parse` returns (which is `-2`) so i was thinking we reserve the `-2` return value for memory allocation error, so it makes more sense to create our own macro.

Reply:
You can handle it after the function returns it and replace it with **ENOMEM**. I don't like for memory allocation error to be that close to 0.

Reply:
Sure thing, ill fix it right away

Reply:
update: There is another problem, the `ini_parse` function returns a `>0` number to specify in which line there was a parsing error. The `ENOMEM` macro has a value of `12` so if i replace the return value with ENOMEM, its specifying that there was a parsing error on the 12 line. I can maybe create a macro that is further than 0? It should be a negative number though, so we can't use `ENOMEM`

Reply:
> Why not `-ENOMEM`, provided to us by the Linux kernel?

As I said `-ENOMEM`.

Reply:
Ah i didn't notice that, my bad.